### PR TITLE
cmake: FFmpeg linking rework

### PR DESCRIPTION
**Pull request by lat9nq**
This adds [FFmpeg](https://ffmpeg.org/) as a submodule and builds/statically links to it on Linux. MSVC is unchanged for the most part. MinGW now uses the same libraries as MSVC, thus fixing the dynamic linking issues with [Martchus](https://github.com/Martchus/PKGBUILDs)'s FFmpeg libraries.

When configuring the submodule on Linux, it disables almost everything from FFmpeg and compiles exactly what yuzu needs. ("Exactly" as far as I know, maybe more can be disabled.) Compile times shouldn't take too much of a hit since yuzu uses only a bit of FFmpeg.

Ports FindFFmpeg.cmake from ctira-emu/citra with minor modifications (see relevant commit). This in turn fixes a couple odd warnings/errors during CMake configuration.

Adds the variable `YUZU_USE_BUNDLED_FFMPEG` to instruct CMake on where to look for FFmpeg. At the moment, it's default to ON. Before this is merged, I want to set it to OFF by default for Linux and ON for WIN32. I'm initially PR'ing this as ON everywhere to let it pass CI that way. Please do not merge until I switch that.

Necessary for future work planned by @ameerj.

### CODE REVIEW

File: src/video_core/command_classes/codecs/codec.cpp [hunk 1]
```diff
@@ -13,7 +13,7 @@
 #include "video_core/memory_manager.h"
 
 extern "C" {
-#include <libavutil/opt.h>
+#include "libavutil/opt.h"
```
Comment (lioncash) [inline src/video_core/command_classes/codecs/codec.cpp:16]: Is this change necessary? We always use `<>` for external libraries.

Comment (lat9nq) [reply src/video_core/command_classes/codecs/codec.cpp:16]: Nah, it's not. I'll fix that.

**Comment by lat9nq** [comment]
Requires yuzu-emu/build-environments#26: we need an assembler to compile FFmpeg on linux-fresh.

**Comment by lat9nq** [comment]
If yuzu-emu/ext-windows-bin#17 gets merged before this PR, I can update this one to point to the new binaries.

**Comment by lat9nq** [comment]
FFmpeg submodule version is set to 4.3.1, and WIN32 now uses the recently uploaded FFmpeg 4.3.1 binaries from yuzu-emu/ext-windows-bin. The external is now configured to be a fallback solution to the system-installed version on Linux. `YUZU_USE_BUNDLED_FFMPEG` is OFF by default on Linux and ON on WIN32. With all of that, it's ready for a final review.

**Review by ameerj — APPROVED** [review]
